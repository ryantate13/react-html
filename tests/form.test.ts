import {form} from '../src';

describe('form', () => {
    it('renders with children', () => {
        expect(form({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(form({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(form({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(form()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(form({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(form({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(form({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(form({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(form({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(form({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(form({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(form({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(form({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(form({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(form({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(form({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(form({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(form({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(form({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(form({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(form({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(form({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(form({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(form({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(form({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(form({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(form({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(form({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(form({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(form({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(form({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(form({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(form({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(form({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(form({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(form({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(form({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(form({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(form({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(form({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(form({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(form({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(form({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(form({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(form({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(form({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(form({translate: "no"})).toMatchSnapshot();
    });

    it('renders with acceptCharset="UTF-8"', () => {
        expect(form({acceptCharset: "UTF-8"})).toMatchSnapshot();
    });

    it('renders with action="https://google.com/"', () => {
        expect(form({action: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with autoComplete="on"', () => {
        expect(form({autoComplete: "on"})).toMatchSnapshot();
    });

    it('renders with autoComplete="off"', () => {
        expect(form({autoComplete: "off"})).toMatchSnapshot();
    });

    it('renders with encType="application/x-www-form-urlencoded"', () => {
        expect(form({encType: "application/x-www-form-urlencoded"})).toMatchSnapshot();
    });

    it('renders with encType="multipart/form-data"', () => {
        expect(form({encType: "multipart/form-data"})).toMatchSnapshot();
    });

    it('renders with encType="text/plain"', () => {
        expect(form({encType: "text/plain"})).toMatchSnapshot();
    });

    it('renders with method="GET"', () => {
        expect(form({method: "GET"})).toMatchSnapshot();
    });

    it('renders with method="POST"', () => {
        expect(form({method: "POST"})).toMatchSnapshot();
    });

    it('renders with method="dialog"', () => {
        expect(form({method: "dialog"})).toMatchSnapshot();
    });

    it('renders with name="TEST"', () => {
        expect(form({name: "TEST"})).toMatchSnapshot();
    });

    it('renders with noValidate=true', () => {
        expect(form({noValidate: true})).toMatchSnapshot();
    });

    it('renders with noValidate=false', () => {
        expect(form({noValidate: false})).toMatchSnapshot();
    });

    it('renders with target="_blank"', () => {
        expect(form({target: "_blank"})).toMatchSnapshot();
    });
});