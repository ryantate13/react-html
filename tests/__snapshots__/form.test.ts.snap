// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form renders arrays of children 1`] = `
<form>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</form>
`;

exports[`form renders with acceptCharset="UTF-8" 1`] = `
<form
  acceptCharset="UTF-8"
/>
`;

exports[`form renders with accessKey="a,b,c,d" 1`] = `
<form
  accessKey="a,b,c,d"
/>
`;

exports[`form renders with action="https://google.com/" 1`] = `
<form
  action="https://google.com/"
/>
`;

exports[`form renders with autoCapitalize="characters" 1`] = `
<form
  autoCapitalize="characters"
/>
`;

exports[`form renders with autoCapitalize="none" 1`] = `
<form
  autoCapitalize="none"
/>
`;

exports[`form renders with autoCapitalize="off" 1`] = `
<form
  autoCapitalize="off"
/>
`;

exports[`form renders with autoCapitalize="on" 1`] = `
<form
  autoCapitalize="on"
/>
`;

exports[`form renders with autoCapitalize="sentences" 1`] = `
<form
  autoCapitalize="sentences"
/>
`;

exports[`form renders with autoCapitalize="words" 1`] = `
<form
  autoCapitalize="words"
/>
`;

exports[`form renders with autoComplete="off" 1`] = `
<form
  autoComplete="off"
/>
`;

exports[`form renders with autoComplete="on" 1`] = `
<form
  autoComplete="on"
/>
`;

exports[`form renders with children 1`] = `
<form>
  TEST
</form>
`;

exports[`form renders with className="foo bar baz" 1`] = `
<form
  className="foo bar baz"
/>
`;

exports[`form renders with contentEditable="false" 1`] = `
<form
  contentEditable="false"
/>
`;

exports[`form renders with contentEditable="true" 1`] = `
<form
  contentEditable="true"
/>
`;

exports[`form renders with dir="auto" 1`] = `
<form
  dir="auto"
/>
`;

exports[`form renders with dir="ltr" 1`] = `
<form
  dir="ltr"
/>
`;

exports[`form renders with dir="rtl" 1`] = `
<form
  dir="rtl"
/>
`;

exports[`form renders with draggable="false" 1`] = `
<form
  draggable="false"
/>
`;

exports[`form renders with draggable="true" 1`] = `
<form
  draggable="true"
/>
`;

exports[`form renders with empty attributes 1`] = `<form />`;

exports[`form renders with encType="application/x-www-form-urlencoded" 1`] = `
<form
  encType="application/x-www-form-urlencoded"
/>
`;

exports[`form renders with encType="multipart/form-data" 1`] = `
<form
  encType="multipart/form-data"
/>
`;

exports[`form renders with encType="text/plain" 1`] = `
<form
  encType="text/plain"
/>
`;

exports[`form renders with hidden=false 1`] = `
<form
  hidden={false}
/>
`;

exports[`form renders with hidden=true 1`] = `
<form
  hidden={true}
/>
`;

exports[`form renders with id="TEST" 1`] = `
<form
  id="TEST"
/>
`;

exports[`form renders with inputMode="decimal" 1`] = `
<form
  inputMode="decimal"
/>
`;

exports[`form renders with inputMode="email" 1`] = `
<form
  inputMode="email"
/>
`;

exports[`form renders with inputMode="none" 1`] = `
<form
  inputMode="none"
/>
`;

exports[`form renders with inputMode="numeric" 1`] = `
<form
  inputMode="numeric"
/>
`;

exports[`form renders with inputMode="search" 1`] = `
<form
  inputMode="search"
/>
`;

exports[`form renders with inputMode="tel" 1`] = `
<form
  inputMode="tel"
/>
`;

exports[`form renders with inputMode="text" 1`] = `
<form
  inputMode="text"
/>
`;

exports[`form renders with inputMode="url" 1`] = `
<form
  inputMode="url"
/>
`;

exports[`form renders with is="foo" 1`] = `
<form
  is="foo"
/>
`;

exports[`form renders with itemID="https://google.com/" 1`] = `
<form
  itemID="https://google.com/"
/>
`;

exports[`form renders with itemProp="https://google.com/" 1`] = `
<form
  itemProp="https://google.com/"
/>
`;

exports[`form renders with itemRef="foo bar" 1`] = `
<form
  itemRef="foo bar"
/>
`;

exports[`form renders with itemScope=false 1`] = `
<form
  itemScope={false}
/>
`;

exports[`form renders with itemScope=true 1`] = `
<form
  itemScope={true}
/>
`;

exports[`form renders with itemType="https://google.com/" 1`] = `
<form
  itemType="https://google.com/"
/>
`;

exports[`form renders with lang="" 1`] = `
<form
  lang=""
/>
`;

exports[`form renders with lang="es-419" 1`] = `
<form
  lang="es-419"
/>
`;

exports[`form renders with method="GET" 1`] = `
<form
  method="GET"
/>
`;

exports[`form renders with method="POST" 1`] = `
<form
  method="POST"
/>
`;

exports[`form renders with method="dialog" 1`] = `
<form
  method="dialog"
/>
`;

exports[`form renders with name="TEST" 1`] = `
<form
  name="TEST"
/>
`;

exports[`form renders with noValidate=false 1`] = `
<form
  noValidate={false}
/>
`;

exports[`form renders with noValidate=true 1`] = `
<form
  noValidate={true}
/>
`;

exports[`form renders with slot="TEST" 1`] = `
<form
  slot="TEST"
/>
`;

exports[`form renders with spellCheck="false" 1`] = `
<form
  spellCheck="false"
/>
`;

exports[`form renders with spellCheck="true" 1`] = `
<form
  spellCheck="true"
/>
`;

exports[`form renders with style={"color":"#000000"} 1`] = `
<form
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`form renders with tabIndex=1 1`] = `
<form
  tabIndex={1}
/>
`;

exports[`form renders with target="_blank" 1`] = `
<form
  target="_blank"
/>
`;

exports[`form renders with title="TEST" 1`] = `
<form
  title="TEST"
/>
`;

exports[`form renders with translate="no" 1`] = `
<form
  translate="no"
/>
`;

exports[`form renders with translate="yes" 1`] = `
<form
  translate="yes"
/>
`;

exports[`form renders without attributes 1`] = `<form />`;
